go/azidext/azure_identity_credential_adapter.go

```diff
@@ -8,18 +8,22 @@ import (
 	"net/http"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/Azure/go-autorest/autorest"
 )
 
-// NewAzureIdentityCredentialAdapter is used to adapt an azcore.Credential to an autorest.Authorizer
-func NewAzureIdentityCredentialAdapter(credential azcore.Credential, options azcore.AuthenticationPolicyOptions) autorest.Authorizer {
-	policy := credential.AuthenticationPolicy(options)
-	return &policyAdapter{p: policy}
+// NewTokenCredentialAdapter is used to adapt an azcore.TokenCredential to an autorest.Authorizer
+func NewTokenCredentialAdapter(credential azcore.TokenCredential, requestOptions policy.TokenRequestOptions) autorest.Authorizer {
```

Comment:
I think we want `scopes []string` instead of `policy.TokenRequestOptions` because the adapter can't function without scopes and `TokenRequestOptions.TenantID` has no effect here.

Reply:
Good idea, fixed.